extra_data/keydata.py

```diff
@@ -156,7 +159,10 @@ def xarray(self, extra_dims=None, roi=()):
         if self.shape[0]:
             coords = {'trainId': self._trainid_index()}
 
-        return xarray.DataArray(ndarr, dims=dims, coords=coords)
+        if name is None:
+            name = f'{self.source}.{self.key}'
```

Comment:
We already make a name for pandas Series objects, with a `/` rather than a `.`. We also remove a trailing `.value` (all control keys have `.timestamp` and `.value` in the file - we should probably limit it to control sources):

https://github.com/European-XFEL/EXtra-data/blob/master/extra_data/keydata.py#L168-L170

Not essential, but it would be good to be consistent between series and xarray objects if it's practical to do so.

Reply:
Coming from live Karabo, I'm all for removing the trailing `.value` given it's a file-only peculiarity. I could see two issues here:

- Somebody might want to merge two arrays with the values and timestamp into a Dataset, which would end up with the same name
- Given file's always show this `.value` and `.timestamp`, a user without knowledge of live Karabo may be confused by it missing

I'm not so keen on breaking from it by joining with a slash. But if the decision has already been made for pandas, I suppose we should stick with it?

Reply:
> Somebody might want to merge two arrays with the values and timestamp into a Dataset, which would end up with the same name

Wouldn't one be `...foo` and the other `...foo.timestamp`?

> Given file's always show this .value and .timestamp, a user without knowledge of live Karabo may be confused by it missing

True enough if they dig into the files. There's definitely a trade-off between being more like the Karabo name and more like the name in the file.

> I'm not so keen on breaking from it by joining with a slash. 

I'm not sure what you mean by breaking it - does Karabo or some other context show them joined with a dot? I was unaware of anything showing source + key in a single string, so I picked a slash basically arbitrarily. I think it should mean that splitting on the last `/` separates source & key, whereas possibly source names could contain `.`?

Reply:
> Wouldn't one be `...foo` and the other `...foo.timestamp`?

We shall not speak of it again. But since we're already here, may I ask for automatically attaching `.value` to control sources, if neither `.value` or `.timestamp` is passed as `key`? Tripped over that way too often.

> I'm not sure what you mean by breaking it - does Karabo or some other context show them joined with a dot? I was unaware of anything showing source + key in a single string, so I picked a slash basically arbitrarily. I think it should mean that splitting on the last `/` separates source & key, whereas possibly source names could contain `.`?

All throughout karabo-gui, e.g. any widget created from a property (tooltip shown):

![kgui-device-property](https://user-images.githubusercontent.com/56224768/89898938-d4a61480-dbe1-11ea-8d36-3fd42b1d50b3.png)

It get's a bit tricky for fast data, as unlike the framework, the karabo-gui always shows a `.` instead of a `:`. I'd wager we might convince CTRL to fix that though.

Reply:
> But since we're already here, may I ask for automatically attaching .value to control sources, if neither .value or .timestamp is passed as key?

Yup, I'll open a new issue for it.

> All throughout karabo-gui, e.g. any widget created from a property

My bad. I've never actually used Karabo much. In that case, I'd say stick with `.` - it's probably more valuable to be consistent with displayed names in Karabo.

Reply:
I've added the removal of a trailing `.value` to `.xarray()` and sneaked a fix in for `.series()`, which technically could have applied it to fast data, too. While it's somewhat unlikely, the key in a pipeline hash _may_ end with `.value`.